pushy/src/main/java/com/turo/pushy/apns/proxy/HttpProxyHandlerFactory.java

```diff
@@ -41,6 +47,53 @@
     private final String username;
     private final String password;
 
+    /**
+     * Checks the system default proxies to determine if an {@link HttpProxyHandlerFactory}
+     * is required for the given host or not, returning an instance or null for each case.
+     *
+     * @param apnsHost the APNs host to check e.g. "api.push.apple.com".
+     * @return an {@link HttpProxyHandlerFactory} if one is required, or <code>null</code> if not.
```

Comment:
> `…or <code>null</code> if not.`

So, two thoughts here. First, as a trivial formatting/style thing, we tend to use `{@code null}` in Javadoc elsewhere in the codebase. While it's certainly no big deal, would you mind doing the same here?

Second, and more significantly, potentially returning `null` feels a little strange to me, though I certainly see where you're going. I suppose the alternative would be to introduce something like a `SystemProxyNotAvailableException` and potentially throw it from this method. What do you think?

Reply:
The first is done, didn't actually know about that, thanks.

The second is API style choice and so depends on what you want your project to look like, I really don't mind. I think the present form is handy though, as my builder code looks like

        HttpProxyHandlerFactory proxyHandlerFactory = HttpProxyHandlerFactory.fromSystemProxies(apnsHost);
        LOGGER.trace("Using proxy handler factory [{}]", proxyHandlerFactory);
        apnsClientBuilder.setProxyHandlerFactory(proxyHandlerFactory);

So at the point I'm building the client, there being a system proxy or not is transparent and works either way.

Reply:
Makes sense. Thanks!